composer/utils/inference.py

```diff
@@ -0,0 +1,166 @@
+        load_object_store (ObjectStore, optional): If the ``load_path`` is in an object name  in a cloud bucket
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used to retreive the checkpoint.
+            Otherwise, if the checkpoint is a local filepath, set to ``None``. (default: ``None``)
+        strict (bool): Whether the keys (i.e., model parameter names) in the model state dict should
+            perfectly match the keys in the model instance. (default: ``False``)
+
+    Returns:
+        None
+    """
+    target_format = ExportFormat(target_format)
+
+    if is_model_deepspeed(model):
+        raise ValueError(f'Exporting for deepspeed models is currently not supported.')
+
+    # Only rank0 exports the model
+    if dist.get_global_rank() != 0:
+        return
+
+    # make a copy of the model so that we don't modify the original model
+    model = copy.deepcopy(model)
+
+    # Apply surgery algorithms in the given order
+    for alg in ensure_tuple(surgery_algs):
+        model = alg(model)
+
+    if load_path is not None:
+        # download checkpoint and load weights only
+        log.debug('Loading checkpoint at %s', load_path)
+        with tempfile.TemporaryDirectory() as tempdir:
+            composer_states_filepath, _, _ = download_checkpoint(path=load_path,
+                                                                 node_checkpoint_folder=tempdir,
+                                                                 object_store=load_object_store,
+                                                                 progress_bar=True)
+            state_dict = torch.load(composer_states_filepath, map_location='cpu')
+            missing_keys, unexpected_keys = model.load_state_dict(state_dict['state']['model'], strict=strict)
+            if len(missing_keys) > 0:
+                log.warning(f"Found these missing keys in the checkpoint: {', '.join(missing_keys)}")
+            if len(unexpected_keys) > 0:
+                log.warning(f"Found these unexpected keys in the checkpoint: {', '.join(unexpected_keys)}")
+
+    model.eval()
+    # Apply transformations (i.e., inference optimizations) in the given order
+    for transform in ensure_tuple(transforms):
+        model = transform(model)
+
+    is_remote_store = store_object_store is not None
+    tempdir_ctx = tempfile.TemporaryDirectory() if is_remote_store else contextlib.nullcontext(None)
+    with tempdir_ctx as tempdir:
+        if is_remote_store:
+            local_store_path = os.path.join(str(tempdir), 'model.export')
+        else:
+            local_store_path = store_path
+
+        if target_format == ExportFormat.TORCHSCRIPT:
+            export_model = None
+            try:
+                export_model = torch.jit.script(model)
+            except Exception as e:
+                log.warning(
```

Comment:
nit: instead of stringifying the exception (which would lose the stack trace), set `exc_info=True` -- see https://docs.python.org/3/library/logging.html#logging.debug